test/SetPath.cpp

```diff
@@ -0,0 +1,421 @@
+/*
+ * uriparser - RFC 3986 URI parsing library
+ *
+ * Copyright (C) 2025, Sebastian Pipping <sebastian@pipping.org>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#undef NDEBUG  // because we rely on assert(3) further down
+
+#include <cassert>
+
+#include <gtest/gtest.h>
+
+#include <uriparser/Uri.h>
+
+namespace {
+
+static void testIsWellFormedPath(const char * candidate, bool hasHost, bool expectedWellFormed) {
+	const char * const first = candidate;
+	const char * const afterLast = (candidate == NULL) ? NULL : (candidate + strlen(candidate));
+
+	const UriBool actualWellFormed = uriIsWellFormedPathA(first, afterLast, hasHost);
+
+	ASSERT_EQ(actualWellFormed, expectedWellFormed);
+}
+
+static UriUriA parseWellFormedUri(const char * text) {
+	UriUriA uri;
+	const int error = uriParseSingleUriA(&uri, text, NULL);
+	// NOTE: we cannot use ASSERT_EQ here because of the outer non-void return type
+	assert(error == URI_SUCCESS);
+	return uri;
+}
+
+static void assertUriEqual(const UriUriA * uri, const char * expected) {
+	int charsRequired = -1;
+	ASSERT_EQ(uriToStringCharsRequiredA(uri, &charsRequired), URI_SUCCESS);
+	ASSERT_TRUE(charsRequired >= 0);
+
+	char * const buffer = (char *)malloc(charsRequired + 1);
+	ASSERT_TRUE(buffer != NULL);
+
+	ASSERT_EQ(uriToStringA(buffer, uri, charsRequired + 1, NULL), URI_SUCCESS);
+
+	ASSERT_EQ(strcmp(buffer, expected), 0);
```

Comment:
I've tried around with the test suite a little more to fully understand `TransformEmptyLeadPathSegments` and the correctness of `SetPath()` in general. While doing so, I learned that this works, and returns better error messages:

```suggestion
	ASSERT_STREQ(buffer, expected);
```

results in:
```
Expected equality of these values:
  buffer
    Which is: "scheme:/./"
  expected
    Which is: "scheme:/"
```
when deleting the removal of the second empty segment.